warehouse/warehouse.go

```diff
@@ -513,22 +515,23 @@ func uploadFrequencyExceeded(warehouse warehouseutils.WarehouseT, syncFrequency
 	return false
 }
 
-func setLastProcessedMarker(warehouse warehouseutils.WarehouseT) {
+func setLastProcessedMarker(warehouse warehouseutils.WarehouseT, lastProcessedTime time.Time) {
 	lastProcessedMarkerMapLock.Lock()
 	defer lastProcessedMarkerMapLock.Unlock()
-	lastProcessedMarkerMap[warehouse.Identifier] = time.Now().Unix()
+	lastProcessedMarkerMap[warehouse.Identifier] = lastProcessedTime.Unix()
```

Comment:
Check with @sivashanmukh for this if this needs to be populated during restarts.